### DISCUSSION TITLE
HomeKit Access Codes

### DISCUSSION THREAD

**Discussion by rednblkx** [rednblkx/HomeKey-ESP32#40] [Ideas]
<img width="128" src="https://www.cultofmac.com/wp-content/uploads/2022/03/keypad_front1.jpg">
Besides HomeKey, the Home Accessory Protocol(HAP) also supports Keypads(like the one above) which lets you control your lock using Access Codes and also gives the ability to add guests(persons not in your home) with their own access codes.

### Example
The assignment of Access Codes looks like this in the Home App

![Home](https://github.com/rednblkx/HomeKey-ESP32/assets/29458132/8c6124d9-a2c9-4130-aef1-fdf1e13c347b)

This is kind of unrelated to HomeKey but i believe it could be useful to some, so let me know what you think about this and if you want to see it implemented onto this project.

**Comment by rednblkx** [comment]
Important Note: This does not mean it could communicate with a keypad like the one shown above since a keypad is just that, a keypad, the actual implementation lies within the lock it connects to and of course each has it’s own logic.

The idea is to just expose this useful functionality to connect it to whatever, in the same spirit as the home key functionality of this project.

**Comment by rednblkx** [comment]
For anyone interested in the technical side of things, I’ve just published the documentation for this [here](https://github.com/rednblkx/homekit-lock-access-code).

**Reply by 5and0** [reply]
Awesome ... I will check it out. Thanks !
